tests/checkpointing/test_model_checkpoint.py

```diff
@@ -513,3 +515,77 @@ def validation_epoch_end(self, outputs):
 
     # check that last one is also the best one
     assert trainer.dev_debugger.checkpoint_callback_history[-1]['epoch'] == len(monitor) - 1
+
+
+def test_checkpoint_repeated_strategy(tmpdir):
+    """
+    This test validates checkpoint can be called several times without increasing internally its global step if nothing run.
+    """
+
+    os.environ['PL_DEV_DEBUG'] = '1'
+
+    checkpoint_callback = ModelCheckpoint(monitor='val_loss', filepath=osp.join(tmpdir, "{epoch:02d}"))
+
+    class ExtendedBoringModel(BoringModel):
+
+        def validation_step(self, batch, batch_idx):
+            output = self.layer(batch)
+            loss = self.loss(batch, output)
+            return {"val_loss": loss}
+
+    model = ExtendedBoringModel()
+    model.validation_step_end = None
+    model.validation_epoch_end = None
+    trainer = pl.Trainer(max_epochs=1,
+                         limit_train_batches=2,
+                         limit_val_batches=2,
+                         limit_test_batches=2,
+                         checkpoint_callback=checkpoint_callback
+                         )
+
+    trainer.fit(model)
+    trainer.test(model)
+    # test only one chechpoint
+    assert str(os.listdir(tmpdir)) == "['epoch=00.ckpt']"
+
+    assert [*model._func_called_count] == ['training_step', 'training_step_end', 'training_epoch_end', 'test_step', 'test_epoch_end']
+
+    assert model.num_calls('training_step') == 2
+    assert model.num_calls('training_step_end') == 2
+    assert model.num_calls('training_epoch_end') == 1
+
+    assert model.num_calls('test_step') == 2
+    assert model.num_calls('test_epoch_end') == 1
+
+    def get_last_checkpoint():
```

Comment:
this is saved in the checkpoint already, right?

Reply:
Not sure. Could you show me where ?